pkg/agent/svid/rotator.go

```diff
@@ -120,6 +120,18 @@ func (r *rotator) rotateSVID(ctx context.Context) (err error) {
 		return err
 	}
 
+	mgrs := r.c.Catalog.KeyManagers()
+	if len(mgrs) > 1 {
+		return errors.New("more than one key manager configured")
+	}
+	keyBytes, err := x509.MarshalECPrivateKey(key)
+	if err != nil {
+		return err
+	}
+	if _, err := mgrs[0].StorePrivateKey(ctx, &keymanager.StorePrivateKeyRequest{PrivateKey: keyBytes}); err != nil {
+		return fmt.Errorf("generate key pair: %v", err)
```

Comment:
This will break compatibility for folks who are running key managers built against the old interface. I think we should log a warning here instead of completely bailing out. Perhaps something like this?
```suggestion
		r.c.Log.Errorf("could not store new agent key pair: %v", err)
		r.c.Log.Warn("Error encountered while storing new agent key pair. Is your KeyManager plugin is up-to-date?")
```

I think ultimately, we should move this whole keymanager interface to something closer to (or matching?) the server keymanager interface. Exposing the private key is definitely a smell, and is not strictly required. I have opened #670 to track this, it will be a big breaking change, we can tackle it for 0.8.

Reply:
+1 agree with this